tools/wine-build-env/tests/test_build_contract.sh

```
@@ -0,0 +1,136 @@
+    >"$tmp/dirty-install.log" 2>&1; then
+    echo "Build contract installed a runner with unbuilt work" >&2
+    exit 1
+fi
+grep -F 'differs from the canonical clean install plan' "$tmp/dirty-install.log" >/dev/null
+
+grep -F 'libkrb5-dev' "$contract/image/Dockerfile" >/dev/null
+grep -F 'libfreetype-dev:i386' "$contract/image/Dockerfile" >/dev/null
+grep -F 'libfontconfig-dev:i386' "$contract/image/Dockerfile" >/dev/null
+grep -F 'tools/wine-build-env/run-build-container.sh full' \
+    "$root/.github/workflows/wine4office-release.yml" >/dev/null
+! grep -F 'apt-get install' "$root/.github/workflows/wine4office-release.yml" >/dev/null
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)